### DISCUSSION TITLE
How to remove some which-key

### DISCUSSION THREAD

**Discussion by tranbinh1604** [LazyVim/LazyVim#3614] [Q&A] [answered]
Hi, i disabled some duplicated keymap as below, but it still show when I press <leader>, how can I also remove along with disabled keymaps

```
local del = vim.keymap.del

 -- already hide when press <leader>
del("n", "<leader>-") -- split win horizontally
del("n", "<leader>|") -- split win vertically 

 -- still show at all
del("n", "<leader><leader>") -- find files (root dir)
del("n", "<leader>/") -- find files (root dir)
del("n", "<leader>,") -- switch to other buffer

```
from `:verbose` I found 

```
n  <Space>     * <Cmd>lua require("which-key").show(" ", {mode = "n", auto = true})<CR>
	Last set from Lua (run Nvim with -V1 for more details)
x  <Space>     * <Cmd>lua require("which-key").show(" ", {mode = "v", auto = true})<CR>
	Last set from Lua (run Nvim with -V1 for more details)
``` 

but If I set `del({ "n", "x" }, "<leader><leader>")` then getting error 
```
vim/keymap.lua:0: E31: No such mapping

# stacktrace:
  - vim/keymap.lua:0 _in_ **del**
  - lua/config/nomap.lua:10
  - lua/config/keymaps.lua:1
  - /LazyVim/lua/lazyvim/config/init.lua:247
  - /LazyVim/lua/lazyvim/config/init.lua:246 _in_ **_load**
  - /LazyVim/lua/lazyvim/config/init.lua:255 _in_ **load**
  - /LazyVim/lua/lazyvim/config/init.lua:183

```

![CleanShot 2024-06-13 at 01 48 03](https://github.com/LazyVim/LazyVim/assets/162952043/c3be69f2-c90b-4cae-8c73-17a8b2c76d04)

**Comment by dpetka2001** [comment, marked as answer]
You should search this repo for where these mappings are being defined. The latter keys are being defined in the telescope spec as `keys` as you can see [here](https://github.com/LazyVim/LazyVim/blob/d82c11f889b03effa5248170ac1b4b4681391f08/lua/lazyvim/plugins/editor.lua#L179-L185). And [here in the docs](https://www.lazyvim.org/configuration/plugins#%EF%B8%8F-adding--disabling-plugin-keymaps) you can see how to disable plugins mappings that are defined under `keys`. 

They are 2 different things and you need to disable each one in their corresponding way.
